textattack/transformations/word_merges/word_merge_masked_lm.py

```diff
@@ -0,0 +1,184 @@
+            inputs = self._encode_text(masked_texts[i : i + self.batch_size])
+            ids = [
+                inputs["input_ids"][i].tolist() for i in range(len(inputs["input_ids"]))
+            ]
+            with torch.no_grad():
+                preds = self._language_model(**inputs)[0]
+
+            for j in range(len(ids)):
+                try:
+                    # Need try-except b/c mask-token located past max_length might be truncated by tokenizer
+                    masked_index = ids[j].index(self._lm_tokenizer.mask_token_id)
+                except ValueError:
+                    replacement_words.append([])
+                    continue
+
+                mask_token_logits = preds[j, masked_index]
+                mask_token_probs = torch.softmax(mask_token_logits, dim=0)
+                ranked_indices = torch.argsort(mask_token_probs)
+                top_words = []
+                for _id in ranked_indices:
+                    _id = _id.item()
+                    token = self._lm_tokenizer.convert_ids_to_tokens(_id)
+                    if utils.is_one_word(token) and not utils.check_if_subword(
+                        token,
+                        self._language_model.config.model_type,
+                        (masked_index == 1),
+                    ):
+                        if mask_token_probs[_id] > self.min_confidence:
+                            top_words.append(token)
+
+                    if len(top_words) >= self.max_candidates:
+                        break
+
+                replacement_words.append(top_words)
+
+            i += self.batch_size
+
+        return replacement_words
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        transformed_texts = []
+        indices_to_modify = list(indices_to_modify)
+        # find indices that are suitable to merge
+        token_tags = [
+            current_text.pos_of_word_index(i) for i in range(current_text.num_words)
+        ]
+        merge_indices = find_merge_index(token_tags)
+        merged_words = self._get_merged_words(current_text, merge_indices)
+        transformed_texts = []
+        for i in range(len(merged_words)):
+            index_to_modify = indices_to_modify[i]
+            word_at_index = current_text.words[index_to_modify]
+            for word in merged_words[i]:
+                word = utils.strip_BPE_artifacts(
+                    word, self._language_model.config.model_type
+                )
+                if word != word_at_index:
+                    temp_text = current_text.delete_word_at_index(index_to_modify + 1)
+                    transformed_texts.append(
+                        temp_text.replace_word_at_index(index_to_modify, word)
+                    )
+
+        return transformed_texts
+
+    def extra_repr_keys(self):
+        return ["masked_lm_name", "max_length", "max_candidates", "min_confidence"]
+
+
+def find_merge_index(token_tags, indices=None):
```

Comment:
This was my developed code. You can try to relax the constraints or discard the function to see whether the generated texts look reasonable.